azurerm/internal/services/devtestlabs/resource_arm_dev_test_global_shutdown_schedule.go

```diff
@@ -0,0 +1,303 @@
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tags"
+	azSchema "github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tf/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmDevTestLabGlobalShutdownSchedule() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmDevTestLabGlobalShutdownScheduleCreateUpdate,
+		Read:   resourceArmDevTestLabGlobalShutdownScheduleRead,
+		Update: resourceArmDevTestLabGlobalShutdownScheduleCreateUpdate,
+		Delete: resourceArmDevTestLabGlobalShutdownScheduleDelete,
+		Importer: azSchema.ValidateResourceIDPriorToImport(func(id string) error {
+			_, err := parse.GlobalScheduleID(id)
+			return err
+		}),
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"location": azure.SchemaLocation(),
+
+			"virtual_machine_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: devtestValidate.GlobalScheduleVirtualMachineID,
+			},
+
+			"enabled": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  true,
+			},
+
+			"daily_recurrence_time": {
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validation.StringMatch(
+					regexp.MustCompile("^(0[0-9]|1[0-9]|2[0-3]|[0-9])[0-5][0-9]$"),
+					"Time of day must match the format HHmm where HH is 00-23 and mm is 00-59",
+				),
+			},
+
+			"timezone": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validate.VirtualMachineTimeZoneCaseInsensitive(),
+			},
+
+			"notification_settings": {
+				Type:     schema.TypeList,
+				Required: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"enabled": {
+							Type:     schema.TypeBool,
+							Required: true,
+						},
+						"time_in_minutes": {
+							Type:         schema.TypeInt,
+							Optional:     true,
+							Default:      30,
+							ValidateFunc: validation.IntBetween(15, 120),
+						},
+						"webhook_url": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+					},
+				},
+			},
+
+			"tags": tags.Schema(),
+		},
+	}
+}
+
+func resourceArmDevTestLabGlobalShutdownScheduleCreateUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).DevTestLabs.GlobalLabSchedulesClient
+	ctx, cancel := timeouts.ForCreateUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	vmID := d.Get("virtual_machine_id").(string)
+	vm, err := parse.GlobalScheduleVirtualMachineID(vmID)
```

Comment:
this should be
```suggestion
	id, err := parse.GlobalScheduleVirtualMachineID(d.Get("virtual_machine_id").(string))
```

Reply:
Updated to `id` instead of `vm` as suggested. However, I left the `vmID` variable intact because I also require it later when constructing the request to create the schedule. I thought that preferable to duplicating `d.Get("virtual_machine_id").(string)` when constructing the request.